pandas-stubs/core/frame.pyi

```diff
@@ -35,19 +35,19 @@ _ErrorType = Literal["raise", "ignore"]
 
 _ListLike = Union[Series, Index, _np.ndarray, Sequence]
 
-_DType = TypeVar("_DType", bound=_np.dtype)
-
 _ColSubsetType = Union[Series, DataFrame, List[_str], _str, _np.ndarray[_np.str_]]
 
 _FunctionLike = Union[_str, Callable]
 
+_TypeLike = Union[_str, _np.dtype, Type[float], Type[_str]]
```

Comment:
This should be
```suggestion
_TypeLike = Union[_str, Type[_np.dtype], Type[float], Type[_str]]
```
right?

Reply:
Nope, check the code:
```iris.astype(np.dtype('str'))```
vs
```iris.astype(np.dtype)```

`Type[_np.dtype]` expects the second one, but it raises `TypeError`

Reply:
While looking into this, I realized our numpy type hierarchy is wrong. So, let's merge it as it is, and then I'll fix the numpy dtype problem.